gen_cert.py

```diff
@@ -652,3 +678,28 @@ def _generate_certificate(
             )
 
         return (download_uuid, verify_uuid, download_url)
+
+    @classmethod
+    def reuse_cert(cls, last_cert, new_course, new_designation):
+        """
+        Checks if cert is initiallly the same as a new cert template
+        created with the same course id and designation.
+
+        Currently, only the course id and designation differentiate new
+        certificates pdfs. HOWEVER there are some elements that change after
+        the cert is instantiated, like grade, so it is assumed this function
+        is called prior to those changes and just before new cert creatation
+        in order to be relevant.
+
+        Message is logged so we can verify if this optimization is useful.
+        """
+        if (last_cert and
+                new_course == last_cert.course_id and
+                new_designation == last_cert.designation):
+            log_msg = 'Reused CertificateGen for {course_id} and ' \
+                '{designation}'.format(
+                    course_id=str(new_course),
+                    designation=str(new_designation),
+                )
+            log.warning(log_msg)
+            return True
```

Comment:
One big oversight here is that we're not returning _anything_ if this `if` check comes back `False`.

Luckily, this still generally works for now, since Python implicitly returns `None` from `void` functions.

But that means this method either returns `True` or `None`, not `True` or `False`,
so a check like `if cert.is_reusable(...) == False:` would lead to incorrect results.

So let's add a `return False` as the final line of this method.